Library/Homebrew/extend/ENV/super.rb

```diff
@@ -68,6 +68,10 @@ def setup_build_environment(formula = nil)
     self["HOMEBREW_INCLUDE_PATHS"] = determine_include_paths
     self["HOMEBREW_LIBRARY_PATHS"] = determine_library_paths
     self["HOMEBREW_DEPENDENCIES"] = determine_dependencies
+    unless formula.nil?
+      self["HOMEBREW_FORMULA"] = formula.name
+      self["HOMEBREW_FORMULA_VERSION"] = formula.version
```

Comment:
Here's my fix: https://github.com/apjanke/brew/commit/bc51fd5944135cea74c91d7d7a5eff97b1f07f27

I'll make this a PR in a bit once I've done some more testing on it.